test/compilable/staticforeach.d

```diff
@@ -638,3 +638,36 @@ auto bug17660(){
     return x;
 }
 static assert(bug17660()==3);
+
+int breakContinueBan(){
+    static assert(!is(typeof({
+        for(;;){
+            static foreach(i;0..1){
+                break;
+            }
+        }
+    })));
+    static assert(!is(typeof({
+        for(;;){
+            static foreach(i;0..1){
+                continue;
+            }
+        }
+    })));
```

Comment:
Tests that should fail to compile are usually placed inside `fail_compilation`, unless new conventions are used now.